tests/api/test_aws.py

```diff
@@ -303,39 +303,109 @@ def get_statements_by_sid(policy_document):
 )
 def test_grant_bucket_access(iam, users, resources):
     bucket_arn = 'arn:aws:s3:::test-bucket'
-    path_arns = [f'{bucket_arn}{resource}' for resource in resources]
+    path_arns_list = []
+    path_arns_object = []
+    if resources:
+        path_arns_list = [f'{bucket_arn}{resource}' for resource in resources]
+        path_arns_object = [f'{bucket_arn}{resource}/*' for resource in resources]
     user = users['normal_user']
     aws.create_user_role(user)
 
-    aws.grant_bucket_access(user.iam_role_name, bucket_arn, 'readonly', path_arns)
+    aws.grant_bucket_access(user.iam_role_name, bucket_arn, 'readonly', path_arns_list)
 
     policy = iam.RolePolicy(user.iam_role_name, 's3-access')
     statements = get_statements_by_sid(policy.policy_document)
 
-    if path_arns:
+    if path_arns_object:
```

Comment:
I'm not so sure about this test, a lot of `if` which mud the water.

I'm not sure what's the best way of writing this test but maybe you can add another parameter with the expected list of resources?

Reply:
Would we not always need an `if` statement to distinguish between the case when paths are specified and the case when they are not? Or would it be preferable to use an `if` statement to define a list of `expected_resources` so the assertion itself does not need to go within an `if` statement?